packages/wallet/backend/tests/transaction/controller.test.ts

```diff
@@ -0,0 +1,202 @@
+      ],
+      listAll: () => mockedTransactionInsertObjs
+    }
+
+    Reflect.set(
+      transactionController,
+      'transactionService',
+      transactionsControllerDepsMocked
+    )
+  }
+
+  beforeAll(async (): Promise<void> => {
+    bindings = await createContainer(env)
+    appContainer = await createApp(bindings)
+    knex = appContainer.knex
+    authService = await bindings.resolve('authService')
+    transactionController = await bindings.resolve('transactionController')
+    prepareControllerDeps()
+  })
+
+  beforeEach(async (): Promise<void> => {
+    await createUser({ ...args, isEmailVerified: true })
+    await createReqRes()
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.stop()
+    await knex.destroy()
+  })
+
+  afterEach(async (): Promise<void> => {
+    await truncateTables(knex)
+  })
+
+  describe('Get list of transactions', () => {
+    it('should return array of transactions', async () => {
+      const { account, walletAddress } = await prepareTransactionDependencies()
+      prepareControllerDeps({
+        walletAddressId: walletAddress.id,
+        accountId: account.id
+      })
+      req.params = {
+        accountId: account.id,
+        walletAddressId: walletAddress.id
+      }
+      req.query = { orderByDate: 'ASC' }
+      await transactionController.list(req, res, next)
+      expect(res.statusCode).toBe(200)
+      expect(res._getJSONData().data).toHaveLength(2)
+      expect(res._getJSONData()).toMatchObject({
+        success: true,
+        message: 'SUCCESS'
+      })
+
+      expect(res._getJSONData().data[0]).toMatchObject({
+        walletAddressId: walletAddress.id,
+        accountId: account.id
+      })
+    })
+
+    it('should return status 400 if the request body is not valid', async () => {
+      req.query = { orderByDate: 'wrong' }
+      await transactionController.list(req, res, (err) => {
+        next()
+        errorHandler(err, req, res, next)
+      })
+
+      expect(next).toHaveBeenCalledTimes(1)
+      expect(res.statusCode).toBe(400)
+      expect(res._getJSONData()).toMatchObject({
+        success: false,
+        message: 'Invalid input'
+      })
+    })
+  })
+
+  describe('ListAll', () => {
+    it('should return array of transactions', async () => {
+      await transactionController.listAll(req, res, next)
+      expect(res.statusCode).toBe(200)
+      expect(res._getJSONData().data).toHaveLength(
+        mockedTransactionInsertObjs.length
+      )
+      expect(res._getJSONData()).toMatchObject({
+        success: true,
+        message: 'SUCCESS'
+      })
+      expect(res._getJSONData().data[0]).toHaveProperty('walletAddressId')
+      expect(res._getJSONData().data[0]).toHaveProperty('accountId')
+      expect(res._getJSONData().data[0]).toHaveProperty('paymentId')
+      expect(res._getJSONData().data[0]).toHaveProperty('assetCode')
+    })
+  })
+  it('should return status 400 if the request body is not valid', async () => {
```

Comment:
could this be grouped with the above test that also validates query params (orderByDate: 'wrong') with nested "it".
In a broader test that goes through invalid query param cases